opentelemetry-jaeger/src/exporter/config/collector/http_client.rs

```diff
@@ -129,6 +129,7 @@ impl CollectorHttpClient {
 }
 
 #[cfg(test)]
+#[allow(dead_code)]
```

Comment:
This is to suppress the clippy warning for the struct not being constructed. Check this for more details: https://github.com/open-telemetry/opentelemetry-rust/actions/runs/8943870592/job/24569519425?pr=1704

Reply:
I think the issue is that this module `test_http_client` is always defined, while used only when the feature `rt-tokio` is enabled. Should be define this module only when the `rt-tokio` is enabled. 

```rust
#[cfg(test)]
#[cfg(feature = "rt-tokio")]
pub(crate) mod test_http_client {
    ..
}
```

Reply:
That's a good idea. Covered this in #1711